test/reducer_test.dart

```diff
@@ -7527,6 +7527,22 @@ main() {
       expect(() => Design.from_json_str(json_str), throwsA(TypeMatcher<IllegalDesignError>()));
     });
   });
+
+  // Github Issue: https://github.com/UC-Davis-molecular-computing/scadnano/issues/677
+  test('adjusting_helices_view_order_should_update_svg_position', () {
+    Helix helix0 = Helix(idx: 0, grid_position: GridPosition(0, 0), group: "foo");
+    Helix helix1 = Helix(idx: 1, grid_position: GridPosition(0, 1), group: "foo");
+    HelixGroup group = HelixGroup(helices_view_order: [0,1], grid: Grid.square);
+    Design design = Design(helices: [helix0, helix1], groups: {"foo": group});
+    AppState state = app_state_from_design(design);
+    Point<num> original_svg_position = state.design.helices[0].svg_position;
+
+    HelixGroup new_group = HelixGroup(helices_view_order: [1,0], grid: Grid.square);
+    AppState new_state = app_state_reducer(state, GroupChange(old_name: "foo", new_name: "foo", new_group: new_group));
+
+    // New svg position should have changed
+    expect(new_state.design.helices[0].svg_position == original_svg_position, false);
```

Comment:
I'm a bit confused. Shouldn't we be testing that `new_state.design.helices[1].svg_position` is equal to `original_svg_position`, since helices 0 and 1 should swap positions in this change?

Maybe we can rename the variable `original_svg_position` to `original_helix0_svg_position` to emphasize which helix it's referring to.

Reply:
Good point. I wasn't sure if the new svg positions would necessarily be just the old ones but swapped but considering how svg_positions are by default calculated from the view order and the view order has been swapped, then of course we should expect the svg position to swap. I have made the change, will work on implementing the fix now.